.eslintrc

```diff
@@ -39,13 +39,51 @@
     ],
     "@typescript-eslint/camelcase": "off",
     "@typescript-eslint/consistent-indexed-object-style": "error",
+    "@typescript-eslint/consistent-type-assertions": "error",
     "@typescript-eslint/consistent-type-definitions": ["error", "interface"],
     "@typescript-eslint/explicit-function-return-type": "off",
     "@typescript-eslint/explicit-module-boundary-types": "off",
+    "@typescript-eslint/member-ordering": [
+      "error",
+      {
+        "default": {
+          "memberTypes": ["field", "constructor", "method", "signature"],
+          "order": "alphabetically"
+        }
+      }
+    ],
     "@typescript-eslint/naming-convention": "off",
+    "@typescript-eslint/no-base-to-string": "error",
+    "@typescript-eslint/no-confusing-non-null-assertion": "error",
+    "@typescript-eslint/no-confusing-void-expression": "error",
     "@typescript-eslint/no-duplicate-imports": "error",
+    "@typescript-eslint/no-dynamic-delete": "error",
+    "@typescript-eslint/no-extraneous-class": "error",
+    "@typescript-eslint/no-implicit-any-catch": "error",
+    "@typescript-eslint/no-invalid-void-type": "error",
+    "@typescript-eslint/non-nullable-type-assertion-style": "off",
+    "@typescript-eslint/no-parameter-properties": "error",
+    "@typescript-eslint/no-require-imports": "error",
     "@typescript-eslint/no-shadow": "error",
+    "@typescript-eslint/no-unnecessary-boolean-literal-compare": "error",
+    "@typescript-eslint/no-unnecessary-condition": "error",
+    "@typescript-eslint/no-unnecessary-qualifier": "off",
+    "@typescript-eslint/no-unnecessary-type-arguments": "off",
+    "@typescript-eslint/no-unnecessary-type-constraint": "error",
     "@typescript-eslint/no-use-before-define": "error",
+    "@typescript-eslint/prefer-includes": "error",
+    "@typescript-eslint/prefer-nullish-coalescing": "error",
+    "@typescript-eslint/prefer-optional-chain": "error",
+    "@typescript-eslint/prefer-readonly": "off",
```

Comment:
I honestly think we should keep this rule.  I personally really like the access modifiers and we definitely criminally underuse `readonly`

Reply:
I think my concern is that this would be a particularly noisy rule. I am also obviously biased by virtue of leaving it off: I'm basically saying "If I have a limited amount of advocacy points to allocate, I personally don't think this rule is a priority for helping us write type-safe code that is less buggy." 

In my mind we already effectively distinguish between `readonly` and not by wrapping not `readonly` properties with `@observable`. Do you think that mental model is incomplete?

Reply:
That's generally true.  I guess I might be a little raw, because I had a bug that would have been caught if I specified some properties as readonly...

Reply:
@timmartin19 mind sharing me that PR/diff so I can be more informed?

Reply:
Yeah, let me track it down.

Reply:
https://github.com/curology/PocketDerm/pull/12364/files

Basically the fix was to wrap them in `@observable` but if I had marked them as `readonly` (because I didn't think they got updated), I would have gotten an error from typescript and known to convert them to observables.

Reply:
Also, totally agree with the bias towards not overwhelming people.  And this rule would be very noisy.  It might be a "let's come back to it" kind of thing.

Reply:
I see. That does seem like a good rule to apply to models to be able to more thoroughly check out usage. 

The only additional thing I'd note is that that rule is a `requires type information` rule. I added an `.eslint.tsconfig.json` file in this repo a little while ago, and this repo is small enough that perf isn't an issue, but I wonder how that would play out in our larger codebase(s).